### ISSUE TITLE
Apply with LinkedIn on - Mobile View

### ISSUE THREAD

**Issue by jontheophilus** [bullhorn/career-portal#61]
If we have Apply with LinkedIn enabled, we should allow the user to either send a desktop link via email (current functionality) or apply using LinkedIn via their mobile browser.

**Comment by jeremy-dobler** [comment]
For clarification, Jon is talking about current functionality on Apple devices.
Currently, the apply modal is not shown on Apple products because we were hiding functionality that did not work on Apple devices(i.e. uploading a resume).

Job Detail Page on Apple device. Notice that you cannot get to Apply modal where LinkedIn functionality is:
![image](https://cloud.githubusercontent.com/assets/13627280/14794507/1fd181ba-0aea-11e6-80d9-a4855feeb54e.png)

Job Detail Page on non-Apple devices. Clicking Apply shows the option to apply by LinkedIn or by uploading a resume:
![image](https://cloud.githubusercontent.com/assets/13627280/14794647/be23a596-0aea-11e6-9811-25e1e09e6252.png)

**Comment by krsween** [comment]
Hey, @jontheophilus —

This is actually by design. I promise we have a good reason for this. On iOS, Safari is a lot more picky about cross origin requests. If we enable the LinkedIn Apply button on those devices, Safari would actually error out on the request for the data. Since this is a core security feature of iOS and has been since iOS5, we won't be able to support mobile safari. I think we allow applications on Chrome on iOS. 

Here is a stack article that goes into greater depth on this issue:

http://stackoverflow.com/questions/29230820/sign-in-with-linkedin-doesnt-trigger-callback-on-ios-safari-when-using-the-js-a

Sorry that this isn't documented as well as it could be.

**Comment by jontheophilus** [comment]
That's a shame, a lot of candidates expect to be able to apply with their mobile phone, and a lot of candidates are going to be running mobile safari. Yes, Chrome on iOS does seem to work, but a very small number of people will have this. According to [monster](http://hiring.monster.com/hr/hr-best-practices/recruiting-hiring-advice/attracting-job-candidates/mobile-job-apply.aspx), 18% of applicants submitted their application via their mobile, and this number is growing.

Could there be a message stating that their browser doesn't support mobile apply and possibly offer a link to Chrome iOS in addition to offering to email the job link? 

Is there any plans to address this functionality or a workaround? Is this something that LinkedIn has to fix with their Javascript lib?

**Comment by krsween** [comment]
Let me check on the messaging. I was the person that implemented the Apply With Linked In feature, but it's been quite a few lines of code since that was implemented. I'll message back in a bit.

I completely agree that this is a really lame limitation of iOS. But it's not one with an easy solution. Since this app is a front end application, we're limited to the client-side APIs we have available to us. Since LinkedIn built their client-side JS apply feature the way that they did and the Safari team sealed off an important mechanism for XSS (cross-site scripting), it won't work with Safari so we would need another entry point that didn't require XSS. 

To provide a little glimpse into one of the options that we vetted: it seems like the simplest approach would be to build this app to include a server-side API (likely Node) which allows us to ensure that all HTTP requests are proxied through that layer and the DOM never executes any XSS. Instead, the DOM sends data to Node and Node sends data back and forth from other APIs (like LinkedIn). This would be a pretty big architectural shift for a single feature, though, so we would need to make sure that it's what the community using this app really needs. There are (of course) several other ways to solve this problem, but this was the quickest one to highlight that we didn't take dropping this functionality lightly.

**tl;dr: I'll check on the messaging. We have some options, but we need to implement them judiciously. iOS Safari's XSS limitation is totally lame (but I'm sure it's very secure).**

**Comment by krsween** [comment]
I was thinking of [this](https://github.com/bullhorn/career-portal/wiki/Downloading-and-Configuring#ios-5-incompatibility). We documented it on our Wiki. Definitely not the same as having a message when I have LinkedIn enabled and I'm on iOS Safari. It would be easy enough to implement that, though, since the logic for hiding/showing that feature is obviously there. 

If you'd like this feature on your CareerPortal you could fork our repo and take a stab at adding it? Feel free to @ me if you have questions and need help.

**Comment by krsween** [comment]
### Update From LinkedIn
We've received an update from the LinkedIn development team that they are aware of the iOS Safari issue and are planning to develop a new version of their Apply With LinkedIn API. We will continue to provide updates as we receive them as well as provide information regarding how that will influence our roadmap.
